be/src/exec/schema_scanner/schema_user_privileges_scanner.cpp

```diff
@@ -0,0 +1,150 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "exec/schema_scanner/schema_helper.h"
+#include "exec/schema_scanner/schema_user_privileges_scanner.h"
+#include "runtime/primitive_type.h"
+#include "runtime/string_value.h"
+//#include "runtime/datetime_value.h"
+
+namespace doris
+{
+
+SchemaScanner::ColumnDesc SchemaUserPrivilegesScanner::_s_tbls_columns[] = {
+    //   name,       type,          size,     is_null
+    { "GRANTEE", TYPE_VARCHAR, sizeof(StringValue), true},
+    { "TABLE_CATALOG", TYPE_VARCHAR, sizeof(StringValue), true},
+    { "PRIVILEGE_TYPE", TYPE_VARCHAR, sizeof(StringValue), false},
+    { "IS_GRANTABLE", TYPE_VARCHAR, sizeof(StringValue), true},
+};
+
+SchemaUserPrivilegesScanner::SchemaUserPrivilegesScanner()
+        : SchemaScanner(_s_tbls_columns,
+                        sizeof(_s_tbls_columns) / sizeof(SchemaScanner::ColumnDesc)),
+        _priv_index(0) {
+}
+
+SchemaUserPrivilegesScanner::~SchemaUserPrivilegesScanner() {
+}
+
+Status SchemaUserPrivilegesScanner::start(RuntimeState *state) {
+    if (!_is_init) {
+        return Status::InternalError("used before initialized.");
+    }
+    RETURN_IF_ERROR(get_new_table());
+    return Status::OK();
+}
+
+Status SchemaUserPrivilegesScanner::fill_one_row(Tuple *tuple, MemPool *pool) {
+    // set all bit to not null
+    memset((void *)tuple, 0, _tuple_desc->num_null_bytes());
+    const TPrivilegeStatus& tbl_priv_status = _priv_result.privileges[_priv_index];
+    // grantee
```

Comment:
```
    {
        void *slot = tuple->get_slot(_tuple_desc->slots()[0]->tuple_offset());
        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
        const std::string* src = &tbl_priv_status.grantee;
        str_slot->len = src->length();
        str_slot->ptr = (char *)pool->allocate(str_slot->len);
        if (NULL == str_slot->ptr) {
            return Status::InternalError("Allocate memcpy failed.");
        }
        memcpy(str_slot->ptr, src->c_str(), str_slot->len);
    }
```

This kind of code block is used everywhere, I think we can extract a common method for it.